dev-docs/RFCs/v6.x/gpu-aggregation-enhancements.md

```diff
@@ -0,0 +1,112 @@
+# RFC: GPU Aggregation enhancements
+
+* **Authors**: Ravi Akkenapally
+* **Date**: Sep 2018
+* **Status**: **Review**
+
+## Problem
+
+`GPUGridAggregator` needs to support min/max/mean operation and ability to aggregate multiple weights. Current version only supports sum operation on a single weights array. This enhancement makes aggregator more scalable and supports use cases such as GridLayer and HexagonLayer, which require aggregation on two different set of weights, with different aggregation operation, one for Color and one for Elevation calculation.
+
+NOTE: `GPUGridAggregator` is currently an experimental API.
+
+
+## Proposed
+
+Multiple weights can be supported by internally maintaining a separate texture (used as render target) for each weights array. For each weights array, we set required blend modes based on requested aggregation operation and perform render calls and collect aggregated data into corresponding texture. We could use `Multiple Render Target` feature to perform single draw for aggregating multiple weights, but WebGL2 **does not** support blending per render target, blend setup is global and applies to all render targets. Users can still use `size` parameter, as described below, to pack more than one weight (up to 3) and aggregate all of them with single draw call, provided, aggregation operation is same for all of them.
+
+
+### Update `run` method
+
+The public API that needs to change is `run()` method. We change input arguments and return object contents. Current style can still be supported as deprecated.
+
+#### weights arguments
+
+Update `run` method to accept weights as an array of objects with following values. Each object will result in one aggregation cycle.
+
+  * id (String) : used to identify the corresponding aggregation buffer and texture in return object. Must be unique for each weights object.
+  * values (Array) : ArrayBufferView with weight values.
+  * size (Number, default: 1, minValue: 1, maxValue: 3) : Size of a single weight instance. Determines how many distinct weights exist in the array.
+  * operation (Enum {SUM, MEAN, MIN or MAX}, default: SUM) : Defines aggregation operation.
+  * needMinMax (Boolean, default: false) : when true additional aggregation steps are performed to calculate total min and max aggregation values.
+  * combineMinMax (Boolean, default: false) : Applicable only when `needMinMax` is set. When true, both min and max values are calculated in single aggregation step, but only does this for first weight (i.e if `size` is 2 or 3, min and max values are calculated only for first weight)
```

Comment:
Seems to have a lot of limitations, is this a TODO/limitation or because that is all that is needed in all cases?

Reply:
Why would one not want combineMinMax in all cases? I assume there are perf reasons or similar, just encouraging another think about simplifying the number of options...

Reply:
The limitation is due to `blendEquationSeparate`, min and max are separately set for both RGB and Alpha, and Alpha channel can contain only one float. If the user has set `size` to 2 or 3 and wants both min and max to be calculated on all of them, this options needs to set to false, updated the text here.